monitor.py

```diff
@@ -1305,16 +1338,30 @@ def parse_entry(self, line: bytes) -> MutableMapping[str, Any]:  # noqa: C901, C
                 self.state['Taxi'] = False
 
             elif event_type == 'navroute':
+                # assume we've failed out the gate, then pull it back if things are fine
+                self._navroute_orig_time = mktime(strptime(entry['timestamp'], '%Y-%m-%dT%H:%M:%SZ'))
+                self._navroute_retries = 10
                 # Added in ED 3.7 - multi-hop route details in NavRoute.json
-                with open(join(self.currentdir, 'NavRoute.json'), 'rb') as rf:  # type: ignore
-                    try:
-                        entry = json.load(rf)
+                if (nv_json := self._parse_navroute_file()) is not None and 'timestamp' in nv_json:
+                    file_time = mktime(strptime(nv_json['timestamp'], '%Y-%m-%dT%H:%M:%SZ'))
 
-                    except json.JSONDecodeError:
-                        logger.exception('Failed decoding NavRoute.json', exc_info=True)
+                    if self._navroute_orig_time - file_time > MAX_NAVROUTE_DISCREPANCY:
```

Comment:
For clarity/ease of reading, this could use a comment something like:

```python
# Check if NavRoute.json probably hasn't been updated since any *previous* Journal NavRoute event.
```
It took me a while to think through the possible game order of writing things to conclude what this check is actually trying to achieve.